src/tests/cmocka/test_sdap_initgr.c

```diff
@@ -195,7 +195,14 @@ struct sdap_get_initgr_state *prepare_state(struct test_sdap_initgr_ctx *ctx,
     }
     assert_non_null(state->dom);
     assert_non_null(recent_dom_info);
-    recent_dom_info->next = NULL;
+    /* GCC complains about a potential dereference of null pointer here.
+     *
+     * It's a false-positive, but in order to have GNULIB manywarnings
+     * enabled, let's just add the check here.
+     */
+    if (recent_dom_info != NULL) {
+        recent_dom_info->next = NULL;
+    }
```

Comment:
Similar to the resolv patch, I think it's better to fail explicitly than silently skip even in case of a false positive.